activerecord/lib/arel/table.rb

```diff
@@ -14,7 +14,8 @@ class << self; attr_accessor :engine; end
     alias :table_name :name
 
     def initialize(name, as: nil, klass: nil, type_caster: klass&.type_caster)
-      @name = name.to_s
+      @name = name
+      @name = @name.to_s unless Nodes::SqlLiteral === @name
```

Comment:
I think we should probably `Arel.arel_node?` this (and make any necessary follow-through changes that forces) -- wdyt?

Reply:
Agree. That was my initial guess: just passing a node to the constructor. Will see what other changes would be required.